diff --git i/CMakeLists.txt w/CMakeLists.txt
index c2e4aa4..c9a1edc 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.16)
+set(CXX_STANDARD 20)
 
 # Change obs-plugintemplate to your plugin's name in a machine-readable format
 # (e.g.: obs-myawesomeplugin) and set 
@@ -19,7 +20,8 @@ set(LINUX_MAINTAINER_EMAIL "cpyarger@gmail.com")
 # Add your custom source files here - header files are optional and only required for visibility
 # e.g. in Xcode or Visual Studio
 add_subdirectory(src/libremidi)
-include_directories(src/libremidi)
+include_directories(src/libremidi
+	"${LIBOBS_INCLUDE_DIR}/../UI/obs-frontend-api")
 target_sources(${CMAKE_PROJECT_NAME}
 	PRIVATE
 		src/utils.cpp
@@ -52,8 +54,7 @@ target_sources(${CMAKE_PROJECT_NAME}
 
 # /!\ TAKE NOTE: No need to edit things past this point /!\
 
-find_package(libobs REQUIRED)
-find_package(obs-frontend-api REQUIRED)
+find_package(LibObs REQUIRED)
 include(external/ObsPluginHelpers.cmake)
 find_package(Qt5Core REQUIRED)
 find_package(Qt5Widgets REQUIRED)
@@ -69,8 +70,8 @@ target_include_directories(${CMAKE_PROJECT_NAME}
 target_link_libraries(${CMAKE_PROJECT_NAME}
 	PRIVATE
 		libremidi
-		OBS::libobs
-		OBS::obs-frontend-api
+		libobs
+		obs-frontend-api
 		Qt5::Core
 		Qt5::Widgets)
 
diff --git i/external/ObsPluginHelpers.cmake w/external/ObsPluginHelpers.cmake
index e1e512c..6d38e9d 100644
--- i/external/ObsPluginHelpers.cmake
+++ w/external/ObsPluginHelpers.cmake
@@ -22,7 +22,7 @@ endif()
 # global include directories to fix "bad" plugin behaviour
 if(DEFINED LIBOBS_INCLUDE_DIR AND NOT TARGET OBS::libobs)
 	message(DEPRECATION "You are using an outdated method of adding 'libobs' to your project. Refer to the updated wiki on how to build and export 'libobs' and use it in your plugin projects.")
-	find_package(libobs REQUIRED)
+	find_package(LibObs REQUIRED)
 	if(TARGET OBS::libobs)
 		set_target_properties(OBS::libobs PROPERTIES IMPORTED_GLOBAL TRUE)
 		message(STATUS "OBS: Using modern libobs target")
@@ -271,4 +271,4 @@ else()
 				VERBATIM)
 		endfunction()
 	endif()
-endif()
\ No newline at end of file
+endif()
diff --git i/src/obs-midi.cpp w/src/obs-midi.cpp
index f1b771f..f43c977 100644
--- i/src/obs-midi.cpp
+++ w/src/obs-midi.cpp
@@ -8,7 +8,8 @@
 #else
 #include <obs-frontend-api/obs-frontend-api.h>
 #endif
-#include "libremidi/include/libremidi/libremidi.hpp"#include <obs-data.h>
+#include "libremidi/include/libremidi/libremidi.hpp"
+#include <obs-data.h>
 #include <utility>
 
 #include "obs-midi.h"
