diff --git i/cmake/modules/Boost.cmake w/cmake/modules/Boost.cmake
index c8f1c39078..34a639db13 100644
--- i/cmake/modules/Boost.cmake
+++ w/cmake/modules/Boost.cmake
@@ -13,7 +13,7 @@ endif()
 # boost::process was introduced first in version 1.64.0,
 # boost::beast::detail::base64 was introduced first in version 1.66.0
 set(MINIMUM_BOOST_VERSION "1.86.0")
-set(_boost_components "system;filesystem;thread;locale;regex;chrono;atomic;date_time;iostreams;nowide")
+set(_boost_components "filesystem;thread;locale;regex;chrono;atomic;date_time;iostreams;nowide")
 if(EMSCRIPTEN)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --use-port=sdl2 --use-port=zlib -fwasm-exceptions")
     #find_package(Boost ${MINIMUM_BOOST_VERSION} REQUIRED )
