diff --git i/beetsplug/rockbox/database.py w/beetsplug/rockbox/database.py
index 8906cba..617d3a0 100644
--- i/beetsplug/rockbox/database.py
+++ w/beetsplug/rockbox/database.py
@@ -289,6 +289,7 @@ class Database:
     def set_list_tag(self, tag_a: str, tag_b: str, item: Item):
         t = item.get(tag_b)
         if t and isinstance(t, list):
+            print("setting tag: ", tag_a, t[0])
             self.add_tag(tag_a, t[0])
 
     def get_tag_seek(
@@ -300,6 +301,8 @@ class Database:
         t = item.id if key else item.get(tag_b)
         if t and isinstance(t, (str, int)):
             return self.tag_data[tag_a][t].seek
+        elif t and isinstance(t, list):
+            return self.tag_data[tag_a][t[0]].seek
         else:
             return self.tag_data[tag_a][self._config.unknown].seek
 
@@ -307,7 +310,7 @@ class Database:
         """Add an items tags to the tag databases"""
         self.set_tag("artist", "artist", item)
         self.set_tag("album", "album", item)
-        self.set_tag("genre", "genre", item)
+        self.set_list_tag("genre", "genres", item)
 
         self.set_tag("title", "title", item, key=True)
 
@@ -354,7 +357,7 @@ class Database:
         index = IndexEntry(
             self.get_tag_seek(item, "artist"),
             self.get_tag_seek(item, "album"),
-            self.get_tag_seek(item, "genre"),
+            self.get_tag_seek(item, "genre", "genres"),
             self.get_tag_seek(item, "title", key=True),
             self.get_tag_seek(item, "filename", key=True),
             self.get_tag_seek(item, "composer"),
diff --git i/test/database_test.py w/test/database_test.py
index 1402d31..8ccf231 100644
--- i/test/database_test.py
+++ w/test/database_test.py
@@ -50,6 +50,16 @@ class TestDatabase(TestHelper):
         assert self.db.index[0].artist == self.db.tag_data["artist"]["Lord Huron"].seek
         assert self.db.index[1].artist == self.db.tag_data["artist"]["AURORA"].seek
 
+    def test_genre(self):
+        self.config["rockbox"]["formats"] = ["flac"]
+        self._database()
+
+        item1 = self.add_track(file="01 Looking Back.opus")
+        self.db.add_item(item1)
+        self.db.sort()
+
+        assert "english" in self.db.tag_data["genre"]
+
     def test_indexentry(self):
         self._database()
         item = self.add_track()
@@ -63,21 +73,21 @@ class TestDatabase(TestHelper):
         assert index.length == 1914
         assert index.mtime != 0
 
-    def test_build_all(self):
-        self._database()
-
-        track1 = self.add_track(file="01 Looking Back.opus")
-        track2 = self.add_track(file="02 Bag of Bones.opus")
-        track3 = self.add_track(file="01 Queendom.opus")
-        track4 = self.add_track(file="02 Forgotten Love.opus")
-
-        self.db.add_item(track1)
-        self.db.add_item(track2)
-        self.db.add_item(track3)
-        self.db.add_item(track4)
-        self.db.sort()
-        self.db.add_index(track1)
-        self.db.add_index(track2)
-        self.db.add_index(track3)
-        self.db.add_index(track4)
-        self.db.write()
+    # def test_build_all(self):
+    #     self._database()
+    #
+    #     track1 = self.add_track(file="01 Looking Back.opus")
+    #     track2 = self.add_track(file="02 Bag of Bones.opus")
+    #     track3 = self.add_track(file="01 Queendom.opus")
+    #     track4 = self.add_track(file="02 Forgotten Love.opus")
+    #
+    #     self.db.add_item(track1)
+    #     self.db.add_item(track2)
+    #     self.db.add_item(track3)
+    #     self.db.add_item(track4)
+    #     self.db.sort()
+    #     self.db.add_index(track1)
+    #     self.db.add_index(track2)
+    #     self.db.add_index(track3)
+    #     self.db.add_index(track4)
+    #     self.db.write()
diff --git i/test/helper.py w/test/helper.py
index 6a6ab76..7dc7db8 100644
--- i/test/helper.py
+++ w/test/helper.py
@@ -108,8 +108,8 @@ class TestHelper:
         item.add(self.lib)
         item.update(values)
 
-        for i in item.formatted():
-            print(i, item._get(i))
+        # for i in item.formatted():
+        #     print(i, item._get(i))
 
         item.move(MoveOperation.COPY)
         item.write()
